text/0000-target-bundles.md

```diff
@@ -0,0 +1,245 @@
+    "system-abi": "C",                                      // NEW: what "system" ABI means
+    "c-abi-kind": "cabi_x86_64",                            // NEW: what C-ABI implementation to use
+
+    // Various stuff adjusting how linkage is done
+    "pre-link-args": ["-Wl,--as-needed", "-Wl,-z,noexecstack"],
+    "post-link-args": [],
+    "pre-link-objects-dll": [],
+    "pre-link-objects-exe": [],
+    "post-link-objects": [],
+    "late-link-args": [],
+    "gc-sections-args": [],                                 // NEW: how to strip sections
+    "rpath-prefix": "$ORIGIN",                              // CHANGED: to allow specifying rpath prefix, null to disable rpath altogether
+    "no-compiler-rt": false,
+    "metadata-section": ".note.rustc",                      // NEW: name of section for metadata
+    "has-frameworks": false,                                // NEW: is concept of frameworks supported?
+    "position-independent-executables": true,               // should become a plain linker argument?
+    "lib_allocation_crate": "alloc_system",
+    "exe_allocation_crate": "alloc_jemalloc",
+    // should become templates? `lib{}.so` is much nicer
+    "dll_prefix": "lib",
+    "dll_suffix": ".so",
+    "exe_suffix": "",
+    "staticlib_prefix": "lib",
+    "staticlib_suffix": ".a",
+
+    // LLVM options
+    "cpu": "x86-64",                                        // CPU of the target
+    "features": "",                                         // LLVM features
+    "relocation_model": "pic",
+    "code_model": "default",
+    "eliminate-frame-pointer": true,
+
+    // Apparently OS X is so much of a special case that even moving all the special cases into
+    // configuration is painful.
+    is_like_osx: false,
+
+    // is_like_solaris is handled by the extra gc-sections key
+    // is_like_msvc is handled by the extra metadata-section, linker-kind, eh-method,
+    // dll-storage-attrs, debug-info keys
+    // is_like_windows is handled by the extra system-abi and c-abi-kind keys
+    // is_like_android is handled by the extra debug-info key
+}
+```
+
+## Distribution of targets
+
+Currently every built-in target is distributed along with the rustc compiler. This is suboptimal,
+because in majority of cases users are interested in targets which rustc can compile for, rather
+than the built-in targets rustc knows about, therefore distributing targets built-in into rustc is
+providing no benefits.
+
+Native libraries and linkers aside, it is obvious there’s little sense in distributing targets
+separately from the standard libraries for them. This RFC proposes to change the distribution so
+the rust-std and rust (not rustc or rust-docs) packages begin including the JSON target description
+for the target which the standard library targets. The target JSONs would get installed in
+`$sysroot/share/rustc/targets/` or a similar directory, and the `RUST_TARGET_PATH` environment
```

Comment:
> On thing to consider is that if we had ${libdir}/rustlib/<target-triple>/target.toml (instead of putting the target name in the file name, we put the file within the folder with the target name, along with the compiled libraries).

I don't think that works very well; see https://github.com/rust-lang/rfcs/pull/1711#discussion_r74292019 for my reply to your other comment suggesting this.